### PR TITLE
Log failed random generation

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Log telemetry error when random id generation fails.

### Motivation

As it results in some events getting dropped, I think it's best to log it as an error.

### Additional Notes

As the SDK is not initialized yet when the id generation fails, we need to wait as well for telemetry. I decided not to store it in the buffer of events but rather in another buffer.

This also includes a fix for the telemetry sample rate that was removed when merging the `DatadogProvider`

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x]  Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Review by fuzzybinary — APPROVED** [review]

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt [hunk 1]
```diff
@@ -112,7 +112,7 @@ class DdSdk(
      * @param kind Error kind.
      */
     @ReactMethod
-    fun telemetryError(message: String, stack: String?, kind: String?, promise: Promise) {
+    fun telemetryError(message: String, stack: String, kind: String, promise: Promise) {
```
Comment (0xnm) [inline packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt:115]: we cannot log any error without a stack? why?

File: packages/core/src/DatadogProvider/Buffer/BoundedBuffer.ts [hunk 2]
```diff
@@ -154,4 +171,33 @@ export class BoundedBuffer extends DatadogBuffer {
...
+        message: string,
+        stack: string,
+        kind: string
+    ) => {
+        if (this.telemetryBuffer[kind]) {
+            this.telemetryBuffer[kind].occurences++;
+        } else {
+            this.telemetryBuffer[kind] = {
+                message,
+                stack,
```
Comment (0xnm) [inline packages/core/src/DatadogProvider/Buffer/BoundedBuffer.ts:185]: but I guess the errors of the same kind may have different stack?

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt (hunk 1 above)
Comment (louiszawadzki) [reply packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt:115]: React Native does not allow optional parameters for iOS methods of native modules.
For now I don't see a case where I won't pass a stack or a kind, if that's the case I'll see what makes more sense, to have another declaration with a different signature or `''` as value.

File: packages/core/src/DatadogProvider/Buffer/BoundedBuffer.ts (hunk 2 above)
Comment (louiszawadzki) [reply packages/core/src/DatadogProvider/Buffer/BoundedBuffer.ts:185]: Yes, but I think it's more important to group errors to understand its impact and prevent sending too many events to telemetry as I don't think the part of the stack that might change is very important to us here (and we don't have the sourcemaps to deobfuscate it).

Comment (0xnm) [reply packages/core/src/DatadogProvider/Buffer/BoundedBuffer.ts:185]: indeed, by adding `stack` we increase cardinality. It is ok for me to have *some* stack as long as it allows to address the real issue.

**Review by 0xnm — APPROVED** [review]

**Pull request merged**